learning_resources/etl/ocw.py

```diff
@@ -0,0 +1,388 @@
+        file_s3_path = "courses" + file_s3_path.split("courses")[1]
+
+    content_json = get_file_content(s3_path, file_s3_path, s3_resource, force_overwrite)
+    if content_json:
+        contentfile_data["content"] = content_json.get("content")
+
+    if image_src:
+        contentfile_data["image_src"] = image_src
+
+    return contentfile_data
+
+
+def transform_run(course_data: dict) -> dict:
+    """Convert ocw course data into a dict for a run"""
+    return {
+        "run_id": course_data["run_id"],
+        "published": True,
+        "instructors": parse_instructors(course_data.get("instructors", [])),
+        "description": course_data.get("course_description"),
+        "year": course_data.get("year"),
+        "semester": course_data.get("term"),
+        "availability": AvailabilityType.current.value,
+        "image": {
+            "url": course_data.get("image_src"),
+            "description": course_data.get("course_image_metadata", {}).get(
+                "description"
+            ),
+        },
+        "level": ", ".join(course_data.get("level", [])),
+        "last_modified": course_data.get("last_modified"),
+        "title": course_data.get("course_title"),
+        "slug": course_data.get("slug"),
+        "url": course_data["url"],
+    }
+
+
+def transform_course(course_data: dict) -> dict:
+    """
+    Transforms a course into our normalized data structure
+
+    Args:
+        course_data (dict): course data
+
+    Returns:
+        dict: normalized learning resource data
+    """  # noqa: D401
+
+    uid = course_data.get("legacy_uid")
+
+    if not uid:
+        uid = course_data.get("site_uid")
+
+    if not uid:
+        log.info(
+            "Skipping %s, both site_uid and legacy_uid missing",
+            course_data["slug"],
+        )
+        return None
+    else:
+        uid = uid.replace("-", "")
+    course_data["run_id"] = uid
+    extra_course_numbers = course_data.get("extra_course_numbers", None)
+
+    if extra_course_numbers:
+        extra_course_numbers = [num.strip() for num in extra_course_numbers.split(",")]
+    else:
+        extra_course_numbers = []
+
+    course_id = f"{course_data.get(PRIMARY_COURSE_ID)}"
+    topics = [
+        {"name": topic_name}
+        for topic_name in list(
+            {
+                topic
+                for topic_sublist in course_data.get("topics", [])
+                for topic in topic_sublist
+            }
+        )
+    ]
+
+    return {
+        "readable_id": course_id,
+        "platform": PlatformType.ocw.value,
+        "title": course_data["course_title"],
+        "departments": course_data.get("department_numbers", []),
+        "resource_content_tags": course_data.get("learning_resource_types", []),
+        "image": {
+            "url": course_data.get("image_src"),
+            "description": course_data.get("course_image_metadata", {}).get(
+                "description"
+            ),
+        },
```

Comment:
Here and in `transform_run`, it would be worth grabbing the `image-alt`, too. Not all OCW courses have it, but some do. (I think particularly the newer ones; Studio UI has a place to set it.)

For example: https://ocw.mit.edu/courses/18-102-introduction-to-functional-analysis-spring-2021/data.json